### ISSUE TITLE
Tmux does not read ~/.tmux.conf

### ISSUE THREAD

**Issue by mattysweeps** [tmux/tmux#1882]
### Issue description

If I build tmux from master, my config file at $HOME/.tmux.conf is not loaded. I can tell because I am using https://github.com/gpakosz/.tmux and the colors of the bottom bar do not change.

I last tested this with `2db9a183620e5362221d2b00190a95aacc4ae90d`

I performed a git bisect, and I believe this commit if the culprit: `c0116b2c5ba7d52509a8ce01dd27c2e3a3d4e2af`

As it does not work on that commit but any prior commit works fine. Looging at the commit it's not obvious as to why this is happening, at least not to me.

Note that the fact that .tmux.conf is a sym link is irrelevant: I tested with a sym link and with an actual file and neither config files were loaded.

### Required information

Please provide the following information:

* tmux version (`tmux -V`).
`tmux next-3.1`
* Platform (`uname -sp`).
`Linux unknown`
* $TERM inside and outside of tmux (`echo $TERM`).
inside
`xterm-256color`
outside
`screen-256color`
* Logs from tmux (`tmux kill-server; tmux -vv new`).
client
https://gist.github.com/mattysweeps/8199720f7653ad86902fe842d7a203e4
server
https://gist.github.com/mattysweeps/e293929ceddcd4020510aa03b3720b5d
out
https://gist.github.com/mattysweeps/bf27b14016e97e46f770086e941646bd

**Comment by nicm** [comment]
Are you sure you killed tmux entirely? Do "tmux kill-server".

**Comment by mattysweeps** [comment]
Yup: I ran tmux kill-server between each iteration and the problem still occurs.

Note that I am running `./configure && make` to build, `./tmux` to test my built tmux, and `/usr/bin/tmux` to compare it to the system installed one. (which is tmux 2.8)

[793f4d89d6a04573825e9fa39fbc301e67c4d300]((https://github.com/tmux/tmux/commit/793f4d89d6a04573825e9fa39fbc301e67c4d300)), the "Merge branch 'obsd-master'" commit before [c0116b2c5ba7d52509a8ce01dd27c2e3a3d4e2af]((https://github.com/tmux/tmux/commit/c0116b2c5ba7d52509a8ce01dd27c2e3a3d4e2af)), works. (I just tested again to be sure)

**Comment by nicm** [comment]
tmux is definitely loading your config, you can see it in the log.

**Comment by nicm** [comment]
There is nothing in your config changing the status line.

**Comment by mattysweeps** [comment]
I gathered logs from both 793f4d8 and c0116b2

793f4d8
client: https://gist.github.com/mattysweeps/ac599acf3ba943ccfadde5c1baf39793
server: https://gist.github.com/mattysweeps/ba05650802bdac91b5e8c81797061565
out: https://gist.github.com/mattysweeps/b7c698a1d4e1117907c24ee37b8d6b24

c0116b2
client: https://gist.github.com/mattysweeps/c0d3001865ad770e897c0fda156f8581
server: https://gist.github.com/mattysweeps/7b909c0b1197fb52c68d6dfbc991f107
out: https://gist.github.com/mattysweeps/fce043e1fa840126d5421067819f5a8c

If I strip the timestamp out and diff the server log, I can see there are some changes
https://gist.github.com/mattysweeps/961bb88f5bbb55ca6b06738a4445ebbd

Note the diff still isn't too useful since many of the lines are different due to memory references
However, real changes start here:
https://gist.github.com/mattysweeps/961bb88f5bbb55ca6b06738a4445ebbd#file-diff-txt-L539

**Comment by mattysweeps** [comment]
Here's a view of the diff after I applied some replacements to make the diff a little better:
https://gist.github.com/mattysweeps/92034300d3e2004cc375eb4d81261b43

- 's/0x[0-f]*/0x0redacted0/g' to not diff on memory address differences 
- 's/pid [0-9]*/pid 000000/g' to not diff on pid differences

**Comment by nicm** [comment]
Context diffs are useless, do diff -u please.
